### DISCUSSION TITLE
Microstepping

### DISCUSSION THREAD

**Discussion by Maxys-original** [bassamanator/Sovol-SV06-firmware#34] [Q&A]
Hi, I saw micro stepping was changed for  X/Y motor drivers to be 128. What does it affect? I have micro steps = 16 in my config, what am I missing by not increasing it to 128?

**Comment by bassamanator** [comment]
Reduces the whine of the motors, and smooths out external walls.

**Reply by Maxys-original** [reply]
Hi mate, did you also have Fine Vertical Artifacts on the outer walls after moving to Klipper? Do changing micro stepping to 128 remove it completely?
It looks to me, these artifacts are nothing to do with micro-stepping, but rather the improper setting of Stealhchop. This thread discusses how to set up TMC 2209 drivers in Kllipper to remove vertical artifacts and decrease the motors' wine. https://www.reddit.com/r/VORONDesign/comments/xbx8pl/having_vfas_on_trident/
I am unfortunately away from home for the next few weeks and cannot test anything myself and it looks to me like you are the only person in the whole Sovol community who can tackle this major print quality issue.
From what I understand we just need to get a datasheet of SV06 motors, use it in the excel sheet provided by TMC(link in that Reddit thread), and get out of it two numbers, which must go into printer.cfg in "[tmc2209 stepper_x]" and  [tmc2209 stepper_y]. Sounds easy enough to do to get rid of VFArtifacts for good, but like I said I cannot do it myself, while still would like to share my findings :)

**Reply by bassamanator** [reply]
Yeah I noticed VFAs before increasing microstepping. Can't say that things are perfect by any means, however, VFAs have been significantly reduced. There are many other factors at play as you might be aware if you've been keeping up with youtuber MirageC.

I'm not sure if the changes I've made are correct. Have a look at the `vfa` branch and let me know. I'm doing some test prints right now either way. I'm not expecting much actually, but we'll see.

**Comment by tbonge** [comment]
I was getting MCU shutdown: Timer too close errors on rare occasions with microsteps set to 128. I changed it to 64 and it hasn't happened since. YMMV.

**Reply by bassamanator** [reply]
Interestingly enough, I've tested microsteps 256 for many hours issue free, however, some movements will cause the mcu to crash because it's unable to generate the steps (for those particular movements). Microsteps 128, on the other hand, has never caused an issue with the stock board.

@tbonge what board are you using?

**Reply by tbonge** [reply]
Stock board printing at 200mm/s with pressure advance enabled. The first few layers will print then it will crash once it gets to the high speed infill. Microsteps 64 seems to work so far and I see no noticeable difference. I just wanted to share incase anyone else gets the MCU shutdown error. I am using the same raspberry pi to print on more than one printer at the same time, but I doubt that would affect the MCU on the printer board.

**Reply by TomSeals** [reply]
@tbonge Im getting the same issue with stock board and a BTT Pi v1.2.1. Getting timer too close errors even on simple models. When it was printing it was extremely quiet and my input shaper gave a recommended 8800k accel at with mzv.

**Reply by bassamanator** [reply]
@TomSeals Printer, config, microsteps?

**Reply by TomSeals** [reply]
This was with your sv06 config. I ended up changing the step count down to 32 for the x and y and enabled stealthchop under 150. I've found the under that speed you really don't see a drop in torque. Left it at 128 for the z and extruder. I was also getting minimum extruder temp faults with your start macro. It was calling for the extruder to be at 0 at points. I put the start macro I have on my ender and it's working fine. Just curious is there a reason you have your retraction setting to 1.2mm? Seems a bit high but maybe that's just relative to the stock setup. Plan on changing the heavy stock tool head over to the mini stealth in the very near future.

**Reply by bassamanator** [reply]
Interestingly enough, I used an Orange pi zero 2W for over a year (RPi 4 otherwise), and somehow didn't experience the timer too close issue with 128 microsteps. I have a note about microsteps under `special considerations`, but I'll amend it to include mention of the BTT Pi as well.

BTW the simplicity of the model plays no part in this error, it's particular printhead moves, I've tested this extensively. But that is not to say that there are no other reasons for the `timer too close` error.

> I was also getting minimum extruder temp faults with your start macro.

Using this printer for 3 years, never experienced this, nor have I had anyone bring this up. Likely variables are not being passed from the slicer to macros. See `adjust your slicer` section.

> It was calling for the extruder to be at 0 at points.

I don't doubt that this was your experience, but again, see note above about variables being passed. Here's the [print_start macro](https://github.com/bassamanator/Sovol-SV06-firmware/blob/065f5d0914789262791a1732205aae3bdecef9da/cfgs/misc-macros.cfg#L133), no where is the extruder temp being set to `0`.

---

I have a cht clone nozzle which "slices" up the filament into 3, so retraction values need to be higher. Everything you see in the repo is used by me, or has been tested/verified by someone (e.g., `skr-mini-e3` branch).

**Reply by TomSeals** [reply]
This was using the orcaslicer profile for 0.4 included in your repo and uncommenting out the lines that are in there for the start gcode in printer settings. also tried simply having print_start and got a immediate bed variable error which I can understand as it wasnt being parsed from the slicer.

**Reply by bassamanator** [reply]
Just checked this to make sure: no adjustments are required to the OS, or other slicer, bundles to make them work.

Not quite sure what you "uncommented", but that likely caused your issue.

**Reply by bassamanator** [reply]
[SV06 0.4 nozzle.orca_printer.txt](https://github.com/user-attachments/files/25728972/SV06.0.4.nozzle.orca_printer.txt)

This are my current OS settings btw, I've been using them exclusively. These print settings are the best on this repo.

I would officially love to upload these to the repo, however, OS doesn't do a full export, meaning some print profile settings are missing: `Standard` and`Strong`.

<img width="448" height="259" alt="image" src="https://github.com/user-attachments/assets/443a8547-c9bc-4e92-b7cc-a96b03ddad94" />

⚠️ The `Fast` speeds are not for everyone, beware.
